sqlmesh/utils/date.py

```diff
@@ -21,6 +21,7 @@
 DATE_INT_FMT = "%Y%m%d"
 
 if t.TYPE_CHECKING:
+    from sqlmesh.core.model.meta import IntervalUnit
```

Comment:
Should IntervalUnit just be moved into this module? Seems like a better place for it anyway + prevents utility package from depending on the core one. Btw, we should do the same to the `Interval` below.